docs/lang/articles/reference.md

```diff
@@ -198,6 +198,24 @@ compile time.
 
 ### Conditional expressions
 
+### Static expressions
+
+```
+static_expression ::= "ti.static(" positional_arguments ")"
+```
+
+Static expressions are expressions that are wrapped by a call to `ti.static()`.
+The `positional_arguments` is evaluated at compile time, and the variables inside it must be compile-time constants.
+
+`ti.static()` receives one or more arguments.
+If one argument it passed to it, it returns the argument.
+If more than one arguments is passed to it, it returns a tuple containing all the arguments in the same order as they are passed.
+
+The static expressions work as hints to trigger many metaprogramming functions in Taichi,
```

Comment:
Hint can be a bit misleading. It's more like a compile-time evaluation mechanism, just like C++'s `constexpr`.

Reply:
`constexpr` is a keyword in C++. `ti.static` is a workaround for not being able to add keywords to Python. IMO "hint" seems a proper word here?

Reply:
I feel like it's a "mechanism", not just a hint. I.e. hint makes me think about `inline` in old-C++, where the compiler can choose to ignore it.